extensions/sql/common-sql/src/main/java/org/eclipse/dataspaceconnector/sql/translation/SqlQueryStatement.java

```diff
@@ -0,0 +1,103 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql.translation;
+
+import org.eclipse.dataspaceconnector.spi.query.Criterion;
+import org.eclipse.dataspaceconnector.spi.query.QuerySpec;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+/**
+ * Maps a {@link QuerySpec} to a single SQL {@code SELECT ... FROM ... WHERE ...} statement. The {@code SELECT ...} part
+ * is passed in through the constructor, and the rest of the query is assembled dynamically, based on the
+ * {@link QuerySpec} and the {@link TranslationMapping}.
+ */
+public class SqlQueryStatement implements QueryTranslator<String> {
+
+    private static final String LIMIT = "LIMIT ? ";
+    private static final String OFFSET = "OFFSET ?";
+    private static final String WHERE_TOKEN = "WHERE";
+    private static final String AND_TOKEN = "AND";
+    private final String selectStatement;
+    private final List<String> whereClauses = new ArrayList<>();
+    private final List<Object> parameters = new ArrayList<>();
+
+    public SqlQueryStatement(String selectStatement) {
+        this.selectStatement = selectStatement;
+    }
+
+    @Override
+    public String toTargetDsl() {
```

Comment:
I think the interface should be renamed `QueryTranslator` --> `QueryStatement`, and since it is a SQL-specific class (it resides in the `common-sql` module), I would argue that statement parameters are always supported. 

However, if - at some point - we want to widen the scope of this, i.e. introduce a super-interface, your proposal would be the way to go.

I will:
- rename the class
- rename the method `toTargetDsl` --> `getQueryAsString`

Reply:
--> removed the `QueryTranslator` interface